### PR TITLE
Update go modules to their latest versions

### PR DESCRIPTION
**Pull request by cfergeau**
github.com/openshift/api and k8s/* were not touched as they usually are
trickier to update. The other openshift modules were updated to the
latest version of the `release-4.7` branch.

### CODE REVIEW

**Comment by guillaumerose** [comment]
```
github.com/code-ready/crc/pkg/crc/config imports
	github.com/spf13/viper imports
	github.com/spf13/afero imports
	io/fs: package io/fs is not in GOROOT (/usr/local/go/src/io/fs)
github.com/code-ready/crc/pkg/crc/config imports
	github.com/spf13/viper imports
	github.com/spf13/afero tested by
	github.com/spf13/afero.test imports
	testing/fstest: package testing/fstest is not in GOROOT (/usr/local/go/src/testing/fstest)
make[1]: *** [vendor] Error 1
make: *** [vendorcheck] Error 2
```

circleci not happy, otherwise always good to update! lgtm.

**Comment by cfergeau** [comment]
Apparently a bug in go versions older than 1.15.10, see https://github.com/spf13/afero/pull/291#issuecomment-807324649 and https://github.com/golang/go/issues/44557
Using afero 1.5.1 instead of 1.6.0 seems to avoid this, so I'll use that for now.

File: packaging/package.sh [hunk 1]
```diff
@@ -24,14 +24,14 @@ binDir="${BASEDIR}/root/Applications/CodeReady Containers.app/Contents/Resources
...
 sign "${binDir}/crc"
-sign "${binDir}/crc-admin-helper-darwin"
+sign "${binDir}/admin-helper-darwin"
 sign "${binDir}/crc-driver-hyperkit"
 
 sign "${BASEDIR}/root/Applications/CodeReady Containers.app"
 
 codesign --verify --verbose "${binDir}/hyperkit"
 
-sudo chmod +sx "${binDir}/hyperkit" "${binDir}/crc-admin-helper-darwin" "${binDir}/crc-driver-hyperkit"
```
Comment (praveenkumar) [inline packaging/package.sh:34]: why these changes are part of this PR?

Comment (cfergeau) [reply packaging/package.sh:34]: I must have messed up the rebase, thanks for catching it!
